src/AppleAccelerate.jl

```diff
@@ -1,23 +1,96 @@
 module AppleAccelerate
 
-using Libdl
+using LinearAlgebra
+using Libdl, LAPACK_jll, LAPACK32_jll
 
-if Sys.isapple()
+# For now, only use BLAS from Accelerate (that is to say, vecLib)
+global const libacc = "/System/Library/Frameworks/Accelerate.framework/Accelerate"
+global const libacc_info_plist = "/System/Library/Frameworks/Accelerate.framework/Versions/Current/Resources/Info.plist"
 
-    try
-        global const libacc = "/System/Library/Frameworks/Accelerate.framework/Accelerate"
-        let accel_lib = dlopen(libacc)
-            dlclose(accel_lib)
+function forward_accelerate(interface::Symbol;
+                            new_lapack::Bool = interface == :ilp64,
+                            clear::Bool = false,
+                            verbose::Bool = false)
+    kwargs = Dict{Symbol,String}()
+    if new_lapack
+        if interface == :ilp64
+            kwargs[:suffix_hint] = "\x1a\$NEWLAPACK\$ILP64"
+        else
+            kwargs[:suffix_hint] = "\x1a\$NEWLAPACK"
+        end
+    else
+        if interface == :ilp64
+            throw(ArgumentError("ILP64 accelerate requires new_lapack"))
         end
-    catch
-        error("Accelerate framework not found.")
     end
+    BLAS.lbt_forward(libacc; clear, verbose, kwargs...)
+end
+
+"""
+    load_accelerate(;clear = true, verbose = false)
 
+Load Accelerate, replacing the current LBT forwarding tables if `clear` is `true`.
+Attempts to load the ILP64 symbols if `load_ilp64` is `true`, and errors out if unable.
+"""
+function load_accelerate(;clear::Bool = true, verbose::Bool = false, load_ilp64::Bool = true)
+    # Silently exit on non-Accelerate-capable platforms
+    @static if !Sys.isapple()
+        return
+    end
+    libacc_hdl = dlopen_e(libacc)
+    if libacc_hdl == C_NULL
+        return
+    end
 
-    include("Array.jl")
-    include("DSP.jl")
-    include("Util.jl")
+    # Check to see if we can load ILP64 symbols
+    if load_ilp64 && dlsym_e(libacc_hdl, "dgemm\$NEWLAPACK\$ILP64") == C_NULL
+        error("Unable to load ILP64 interface from '$(libacc)'; are you running macOS 13.3+?")
+    end
```

Comment:
I'm not sure if we have this facility in Julia to query the MacOS version - but it would be nice to say: `You are running MacOS version x.y; you need 13.3+`.

Reply:
Heh, I already wrote that as a part of this PR!  I can integrate it here.